### PR TITLE
add auto-generated doc links

### PR DESCRIPTION
**Pull request by t-bltg**
Fix https://github.com/JuliaParallel/MPI.jl/issues/651.

```julia
julia> using MPI
help?> MPI.API.MPI_Wait
  MPI_Wait(request, status)

    •  MPI_Wait man page https://www.mpich.org/static/docs/latest/www3/MPI_Wait.html
```

Needs https://github.com/JuliaInterop/Clang.jl/pull/386.

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -73,8 +73,7 @@ makedocs(
             "reference/advanced.md",
         ],
         "refindex.md",
-    ],
-    strict = true,
```
Comment (giordano) [inline docs/make.jl:77]: I'm not happy at all about this: #569.

**Review by giordano — CHANGES_REQUESTED** [review]

File: docs/make.jl (hunk 1 above)
Comment (t-bltg) [reply docs/make.jl:77]: Should we add an `API` page in the docs listing all the low level methods ?
How can I detect them automatically ?
It is not conceivable to do it by hand.

Comment (giordano) [reply docs/make.jl:77]: Ask Documenter to collect the functions in a given module: https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/7721c30465a397cba49ee75a7647c4fbac403d96/docs/src/index.md#functions

File: gen/src/MPIgenerator.jl [hunk 2]
```diff
@@ -19,6 +19,13 @@ module MPIgenerator
 
         options = load_options(joinpath(@__DIR__, "generator.toml"))  # wrapper generator options
 
+        doc_url = if MPIPreferences.binary == "MPICH_jll"
+            fname -> "- `$fname` man page `https://www.mpich.org/static/docs/latest/www3/$fname.html`"
+        else
+            fname -> "- `$fname` man page `https://www.open-mpi.org/doc/current/man3/$fname.3.php`"
+        end
+        options["general"]["callback_documentation"] = node -> String[doc_url(node.id)]
```
Comment (simonbyrne) [inline gen/src/MPIgenerator.jl:27]: why not link to both?

Comment (simonbyrne) [reply gen/src/MPIgenerator.jl:27]: Also, you can use  https://github.com/JuliaParallel/MPI.jl/blob/4e8dcd03fb428ee37abb963052cd6009a9aed45b/src/MPI.jl#L30-L34

File: docs/make.jl (hunk 1 above)
Comment (t-bltg) [reply docs/make.jl]: Fixed, by adding `api.md`.

File: gen/src/MPIgenerator.jl (hunk 2 above)
Comment (t-bltg) [reply gen/src/MPIgenerator.jl]: Done, both `OpenMPI` and `MPICH`.

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**